kubetest/e2e.go

```diff
@@ -392,8 +394,15 @@ func DumpClusterLogs(location string) error {
 }
 
 func DumpFederationLogs(location string) error {
-	log.Printf("Dumping Federation logs to: %v", location)
-	return finishRunning(exec.Command("./federation/cluster/log-dump.sh", location))
+	logDumpPath := "./federation/cluster/log-dump.sh"
+	// federation/cluster/log-dump.sh only exists in the Kubernetes tree
+	// post-1.6. If it doesn't exist, do nothing and do not report an error.
+	if i, err := os.Stat(logDumpPath); err == nil && !i.IsDir() {
```

Comment:
Shouldn't you actually check that this is a regular file `i.Mode().IsRegular()`?